### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by MudasirSherwani**
### Required code work is done described below: 🎯 

- [ ] rails g migration model User
- [ ] rails g migration model Like
- [ ] rails g migration model Comment
- [ ] rails g migration model Post
- [ ] Add belongs_to Associations
- [ ] Add has_many Associations
- [ ] Generate schema
- [ ] Config PostgreSQL database
- [ ] rails db:migrate
- [ ] Update README file
- [ ] Fix Linters

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,7 +174,7 @@ Example:
 
 ## 🔭 Future Features <a name="future-features"></a>
 
-- **Add more Classes**
+- **REST Full API**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (Kamba56) [inline README.md:180]: - [ ] The MIT license link in your readme file is displaying a 404 error. Kindly fix the link to your license file.

File: db/migrate/20230329175616_create_like.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateLike < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.belongs_to :users, null: false, foreign_key: true
```
Comment (Kamba56) [inline db/migrate/20230329175616_create_like.rb:4]: - [ ] From the ERD diagram, likes belongs to both users and posts. Kindly create a new migration to add a foreign key that relates likes and posts please.

File: db/migrate/20230329184329_create_post.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :Title
+      t.text :Text
+      t.integer :CommentsCounter
+      t.integer :LikesCounter
+
```
Comment (Kamba56) [inline db/migrate/20230329184329_create_post.rb]: - [x] I don't see any migration where you add the foreign key that relates posts table and users table. You should use migrations to add columns, and foreign keys, create tables, and alter tables. Avoid directly editing the schema file. Kindly add a migration to create a foreign key that relates the users table and the posts.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @MudasirSherwani, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job using migrations to create tables!👏👏👏
- [x] You have done a great job setting up your rails project with PostgreSQL

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @MudasirSherwani 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**